### ISSUE TITLE
Crash when using ResourceLoader.load_threaded_request with multithreaded 3d physics

### ISSUE THREAD

**Issue by lyuma** [godotengine/godot#69186]
### Godot version

4.0.beta6

### System information

Windows 10, and Linux (steam deck)

### Issue description

Godot crashes when loading resources using the multithreaded resource loader.
An example of how the output log will appear
```
   At: Cannot retrieve debug info outside the main thread. Thread ID: -4159884055373798805
background_loader_task_done: res://addons/vsk_menu/ingame_menu/ingame_gui.tscn, error: 0 resource_path: res://addons/vsk_menu/ingame_menu/ingame_gui.tscn
background_loader_thread_ended (success)
GOT a finished signal!
GOT a finished signal!
ERROR: Invalid Group ID
   at: (core/object/worker_thread_pool.cpp:375)
GOT a finished signal!
GOT a finished signal!
GOT a finished signal!
GOT a finished signal!
GOT a finished signal!
[1]+  Segmentation fault      ~/Downloads/Godot_v4.0-beta6_win64.exe
```

May be related to #63671 - I'm not sure. I wanted to submit a new issue since I have a reproduction project, albeit a bit complex.
I was able to reproduce this on beta2 and later. From some commit log messages in my project where the setting was toggled off, crashes existed even before August (alpha 15?), maybe much further back.

Happens more reliably on Linux (I used steam deck), but recently windows reproduces within a few tries too.

Seems to randomly crash in either 2d and 3d physics.

Stack trace when it crashes in 3d physics:
```
>	[Inline Frame] godot.windows.editor.x86_64.exe!memdelete(WorkerThreadPool::BaseTemplateUserdata *) Line 109	C++
 	godot.windows.editor.x86_64.exe!WorkerThreadPool::_add_group_task(const Callable & p_callable, void(*)(void *, unsigned int) p_func, void * p_userdata, WorkerThreadPool::BaseTemplateUserdata * p_template_userdata, int p_elements, int p_tasks, bool p_high_priority, const String & p_description) Line 303	C++
 	[Inline Frame] godot.windows.editor.x86_64.exe!WorkerThreadPool::add_template_group_task(GodotStep3D *) Line 181	C++
 	godot.windows.editor.x86_64.exe!GodotStep3D::step(GodotSpace3D * p_space, float p_delta) Line 366	C++
 	godot.windows.editor.x86_64.exe!GodotPhysicsServer3D::step(float p_step) Line 1645	C++
 	godot.windows.editor.x86_64.exe!PhysicsServer3DWrapMT::thread_step(float p_delta) Line 41	C++
 	godot.windows.editor.x86_64.exe!CommandQueueMT::_flush() Line 374	C++
 	[Inline Frame] godot.windows.editor.x86_64.exe!CommandQueueMT::wait_and_flush() Line 414	C++
 	[Inline Frame] godot.windows.editor.x86_64.exe!PhysicsServer3DWrapMT::thread_loop() Line 59	C++
 	godot.windows.editor.x86_64.exe!PhysicsServer3DWrapMT::_thread_callback(void * _instance) Line 47	C++
 	godot.windows.editor.x86_64.exe!Thread::callback(Thread * p_self, const Thread::Settings & p_settings, void(*)(void *) p_callback, void * p_userdata) Line 67	C++
 	[External Code]	
 	godot.windows.editor.x86_64.exe!thread_start<unsigned int (__cdecl*)(void *),1>(void * const parameter) Line 97	C++
 	[External Code]	
```

Stack trace when it crashes in 2d physics:
```
>	[Inline Frame] godot.windows.editor.x86_64.exe!memdelete(WorkerThreadPool::BaseTemplateUserdata *) Line 109	C++
 	godot.windows.editor.x86_64.exe!WorkerThreadPool::_add_group_task(const Callable & p_callable, void(*)(void *, unsigned int) p_func, void * p_userdata, WorkerThreadPool::BaseTemplateUserdata * p_template_userdata, int p_elements, int p_tasks, bool p_high_priority, const String & p_description) Line 303	C++
 	[Inline Frame] godot.windows.editor.x86_64.exe!WorkerThreadPool::add_template_group_task(GodotStep2D *) Line 181	C++
 	godot.windows.editor.x86_64.exe!GodotStep2D::step(GodotSpace2D * p_space, float p_delta) Line 242	C++
 	godot.windows.editor.x86_64.exe!GodotPhysicsServer2D::step(float p_step) Line 1284	C++
 	godot.windows.editor.x86_64.exe!PhysicsServer2DWrapMT::thread_step(float p_delta) Line 41	C++
 	godot.windows.editor.x86_64.exe!CommandQueueMT::_flush() Line 374	C++
 	[Inline Frame] godot.windows.editor.x86_64.exe!CommandQueueMT::wait_and_flush() Line 414	C++
 	[Inline Frame] godot.windows.editor.x86_64.exe!PhysicsServer2DWrapMT::thread_loop() Line 59	C++
 	godot.windows.editor.x86_64.exe!PhysicsServer2DWrapMT::_thread_callback(void * _instance) Line 47	C++
 	godot.windows.editor.x86_64.exe!Thread::callback(Thread * p_self, const Thread::Settings & p_settings, void(*)(void *) p_callback, void * p_userdata) Line 67	C++
 	[External Code]	
 	godot.windows.editor.x86_64.exe!thread_start<unsigned int (__cdecl*)(void *),1>(void * const parameter) Line 97	C++
 	[External Code]	
```

### Steps to reproduce

1. Open the project once in editor to let things import.
2. Run the project, such as in a debugger.
3. It will show a black screen. I messed up something with the viewport when creating the MRP. It does not matter: the crash still happens.
4. If it does not crash instantly, press escape repeatedly. Escape reloads the menu.
5. Now edit project settings and disable multithreading on 3d physics.
6. The crash will not happen, no matter how many times you try.
7. Note that while the stacktrace sometimes includes 2d physics, disabling 2d physics has no effect. it seems caused by 3d physics for me.

### Minimal reproduction project

[mtcrash.zip](https://github.com/godotengine/godot/files/10095272/mtcrash.zip)

**Comment by lyuma** [comment]
Seems to be fixed by #67680

**Comment by lyuma** [comment]
Duplicate of #67333
